### ISSUE TITLE
ModuleNotFoundError: No module named 'yaml'

### ISSUE THREAD

**Issue by nalzok** [google/flax#2190]
Provide as much information as possible. At least, this should include a description of your issue and steps to reproduce the problem. If possible also provide a summary of what steps or workarounds you have already tried.

### System information
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): GCP TRC TPU VM v3-8
```Linux t1v-n-5d019513-w-0 5.13.0-1023-gcp #28~20.04.1-Ubuntu SMP Wed Mar 30 03:51:07 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux```
- Flax, jax, jaxlib versions (obtain with `pip show flax jax jaxlib`:
```Name: flax
Version: 0.5.1
Summary: Flax: A neural network library for JAX designed for flexibility
Home-page: https://github.com/google/flax
Author: Flax team
Author-email: flax-dev@google.com
License: UNKNOWN
Location: /home/qys/Research/clipper/.venv/lib/python3.10/site-packages
Requires: jax, matplotlib, msgpack, numpy, optax, rich, typing-extensions
Required-by:
---
Name: jax
Version: 0.3.13
Summary: Differentiate, compile, and transform Numpy code.
Home-page: https://github.com/google/jax
Author: JAX team
Author-email: jax-dev@google.com
License: Apache-2.0
Location: /home/qys/Research/clipper/.venv/lib/python3.10/site-packages
Requires: absl-py, numpy, opt-einsum, scipy, typing-extensions
Required-by: chex, flax, optax
---
Name: jaxlib
Version: 0.3.10
Summary: XLA library for JAX
Home-page: https://github.com/google/jax
Author: JAX team
Author-email: jax-dev@google.com
License: Apache-2.0
Location: /home/qys/Research/clipper/.venv/lib/python3.10/site-packages
Requires: absl-py, flatbuffers, numpy, scipy
Required-by: chex, optax
```
- Python version: `Python 3.10.4`
- GPU/TPU model and memory: `TPU v3-8`
- CUDA version (if applicable): N/A

### Problem you have encountered:

Cannot import Flax.

### What you expected to happen:

Can import Flax.

### Logs, error messages, etc:

```
Python 3.10.4 (main, May 10 2022, 00:10:12) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import flax
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/qys/Research/clipper/.venv/lib/python3.10/site-packages/flax/__init__.py", line 19, in <module>
    from . import linen as linen
  File "/home/qys/Research/clipper/.venv/lib/python3.10/site-packages/flax/linen/__init__.py", line 20, in <module>
    from .activation import (
  File "/home/qys/Research/clipper/.venv/lib/python3.10/site-packages/flax/linen/activation.py", line 22, in <module>
    from flax.linen.module import compact
  File "/home/qys/Research/clipper/.venv/lib/python3.10/site-packages/flax/linen/module.py", line 39, in <module>
    from flax.linen import summary
  File "/home/qys/Research/clipper/.venv/lib/python3.10/site-packages/flax/linen/summary.py", line 26, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
```

### Steps to reproduce:
Whenever possible, please provide a *minimal example*. Please consider submitting it as a Colab link.

Here is the `Pipfile`, in case you use [pipenv](https://pipenv.pypa.io/en/latest/).

```
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
clip = {git = "https://github.com/openai/CLIP.git"}
click = "*"
numpy = "*"
jax = "*"
flax = "*"

[dev-packages]

[requires]
python_version = "3.10"
```

See also my `Pipfile.lock`: [Pipfile.lock.txt](https://github.com/google/flax/files/8886867/Pipfile.lock.txt)

**Comment by cgarciae** [comment]
Thank @nalzok for the report! Seems we need to make our dependency on `PyYAML` explicit. I'll push a fix soon.
In the mean time you can fix this by adding `PyYAML>=6.0` to your dependencies.

**Comment by cgarciae** [comment]
Fixed by #2193

**Comment by cgarciae** [comment]
Fix available in `0.5.2`.
